### PR TITLE
Improved support for nested code blocks

### PR DESCRIPTION
**Pull request by nthiery**
Bug fix from @ocayrol, tests adapted from rst2ipynb.

Note: the output could possibly be further improved.

### CODE REVIEW

File: tests/code_blocks.rst [hunk 1]
```diff
@@ -167,3 +167,100 @@ continuation prompt:
...
+
+       1+1
+
+#.  Nested code block B
+
+    ::
+
+       1+1
+
+#. Nested code block C; ok to fail? (the inner indent does not match the itemized text indent)
```
Comment (mmcky) [inline tests/code_blocks.rst]: @nthiery do you think this edge case should be caught and rendered?

File: tests/code_blocks.rst [hunk 2]
```diff
@@ -167,3 +167,100 @@ continuation prompt:
...
+            >>> plus_grand_element([7,3,1,10,4,10,2,9])
+            10
+
+        Foo.
+
+        Bla::
+
+            >>> 1+1
+
+        ok to fail? (missing mandatory new line after `::`)::
```
Comment (mmcky) [inline tests/code_blocks.rst]: I am ok for this to fail as it is part of the `rst` specification for a newline.

**Comment by mmcky** [comment]
thanks @nthiery and @ocayrol this is a nice addition.

**Comment by nthiery** [comment]
I don't really have an opinion for how far to support litigious code-blocks. Presumably the best is to just leave the decision to Sphinx, and just double check that the produced output is consistent with the other formats.

Btw: thanks for your quick feedback and integration!

**Comment by mmcky** [comment]
Thanks @nthiery I will merge this as is. I'm in favour of conforming to RST spec to keep things simple.
